src/Features/Core/Portable/ExternalAccess/UnitTesting/API/INewUnitTestingIncrementalAnalyzerImplementation.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
+{
+    internal interface INewUnitTestingIncrementalAnalyzerImplementation
+    {
+#if false // Not used in unit testing crawling
+        // Task NewSolutionSnapshotAsync(Solution solution, CancellationToken cancellationToken);
+        // Task DocumentOpenAsync(Document document, CancellationToken cancellationToken);
+        // Task DocumentCloseAsync(Document document, CancellationToken cancellationToken);
+        // Task DocumentResetAsync(Document document, CancellationToken cancellationToken);
+        // void RemoveProject(ProjectId projectId);
+
+        // [Obsolete]
+        // bool NeedsReanalysisOnOptionChanged(object sender, UnitTestingOptionChangedEventArgsWrapper e);
+#endif
+
+        Task AnalyzeSyntaxAsync(Document document, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken);
```

Comment:
This also appears to be unused - http://index/?rightProject=Microsoft.CodeAnalysis.UnitTesting.SourceBasedTestDiscovery.Core&file=IncrementalAnalyzer.cs&line=160

Reply:
oooh, hooray.

Reply:
looks like you don't use bodyOpt in AnalyzeDocument either, right?

Reply:
if not, i will remove bodyOpt in my followup pr that cleans things out and removes unncessary parts that aren't actually needed for unit-testing.

Reply:
Yes, looks like bodyOpt is unused. Just curious - what does that represent? Is it providing more info about exactly what (node) changed in the document?

Reply:
correct.  it tells you if the edit was contained entirely to a method body.

Reply:
Interesting... I thought there was another flag that we were relying on for that. Would it be too much work to preserve bodyOpt. I think it would be nice to avoid reanalysis to discover (static) tests when changes are intra-method body.

Reply:
Alternatively, if you could introduce an enum to convey whether the change was intra-method body (without returning the node) that may be nice too.

Reply:
it's a possibility.  though i'm very wary of this because of maintenance costs :)  if you guys want to own that, then that's ok with me :)

Reply:
Ok I think I would vote to just leave it as is then with the assumption that the level of support will remain at whatever was working before :)